compiler/rustc_ty_utils/src/consts.rs

```diff
@@ -1,3 +1,6 @@
+#![deny(rustc::untranslatable_diagnostic)]
+#![deny(rustc::diagnostic_outside_of_impl)]
```

Comment:
Just for clarification, I assume these aren't all the diagnostics in `rustc_ty_utils` and you're just starting with the `rustc_ty_utils::consts` and `rustc_ty_utils::needs_drop` modules? That's perfectly okay, I just want to make sure, because if these are all of them then we can put these `deny` attributes at the root of the crate.

Reply:
I am pretty sure this PR covers all the diagnostics in `rustc_ty_utils` as far as I can tell. The only thing I'm not sure about are the calls to `bug!()`, `span_bug!()`, and `tcx.sess.delay_span_bug()` throughout the crate, but these didn't trigger the `deny` attributes and I got the impression these were for internal compiler bug reporting rather than diagnostics.

If these need translating as well then I can work on those, otherwise I could open a PR to move the `deny` attributes to the root if you want.

Reply:
We can ignore the bug-related functions. If that's everything then you can move the deny attribute to the root - no need to open a PR, just add to this one and I'll re-approve.